types/index.d.ts

```diff
@@ -0,0 +1,4 @@
+export function HomebaseProvider(props?: object): object;
```

Comment:
I think we can get a little more specific with the types. E.g.
- Transaction is an array of objects
- Schema is an object of objects (maybe even include the `type: 'ref'` and `unique: 'identity'` bits as Enums)
- HomebaseProvider takes a `{props?: {config?: {schema?: Schema, initialData?: Transaction }}}`